### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by rodrigoWolox**
## Summary
Tictactoe now using redux. Please dont spear me for the comments in the actions.js file. For now is working with this but i would like to do it like you tell me yesterday, comming on next PRs...

## Screenshot
actions

## Trello Card
https://trello.com/c/MKoOhnZn/14-redux-i-actions-reducers-store-todo-finger

### CODE REVIEW

File: rh-tictactoe/src/app/screens/Game/index.js [hunk 1]
```diff
@@ -4,34 +4,23 @@ import { connect } from 'react-redux';
...
+      stepNumber={props.stepNumber}
+    />
+  );
 }
 
-Game.propTypes = {
+GameContainer.propTypes = {
   handleClick: PropTypes.func.isRequired,
-  history: PropTypes.arrayOf(PropTypes.object),
+  history: PropTypes.arrayOf(PropTypes.shape({ squares: PropTypes.array })),
```
Comment (guidoprinc) [inline rh-tictactoe/src/app/screens/Game/index.js]: array of what? We always are specific to know where the problem can be in the future and make the code better

File: rh-tictactoe/src/app/screens/Game/layout.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+    </div>
+  );
+}
+
+export default Game;
+
+Game.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  history: PropTypes.arrayOf(PropTypes.shape({ squares: PropTypes.array })),
```
Comment (guidoprinc) [inline rh-tictactoe/src/app/screens/Game/layout.js]: same

File: rh-tictactoe/src/index.js [hunk 3]
```diff
@@ -3,12 +3,12 @@ import ReactDOM from 'react-dom';
...
 
 import { store } from './redux/store';
-import Game from './app/screens/Game';
+import GameContainer from './app/screens/Game';
 import './scss/index.scss';
 
 const App = () => (
   <Provider store={store}>
-    <Game />
+    <GameContainer />
```
Comment (guidoprinc) [inline rh-tictactoe/src/index.js]: you can still call it Game because it's a default export

File: rh-tictactoe/src/redux/game/reducer.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import * as utils from '../../utils/utils';
+
+const initialState = {
+  history: [{ squares: Array(9).fill(null) }],
+  xIsNext: true,
+  status: 'Next player: X',
+  stepNumber: 0,
+  endGame: false,
+  moves: []
```
Comment (guidoprinc) [inline rh-tictactoe/src/redux/game/reducer.js]: delete unused states

File: rh-tictactoe/src/utils/utils.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        <button onClick={() => this.props.jumpTo(stepNumber)}>{desc}</button>
+      </li>
+    );
+    return li;
+  });
+  return moves;
+}; */
+
+export function calculateWinner(squares) {
+  let winner = null;
```
Comment (guidoprinc) [inline rh-tictactoe/src/utils/utils.js]: set it on false and then you can use array properties like `some` to make the job easier. We can check this later

File: rh-tictactoe/src/app/screens/Game/components/Moves/index.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  render() {
+    return <ol>{this.props.moves.map(this.renderMove)}</ol>;
+  }
+}
+
+export default Moves;
+
+Moves.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  moves: PropTypes.arrayOf(PropTypes.object)
```
Comment (guidoprinc) [inline rh-tictactoe/src/app/screens/Game/components/Moves/index.js]: mmmmm what did we talk about this? ajhahahha

File: rh-tictactoe/src/app/screens/Game/index.js [hunk 7]
```diff
@@ -1,88 +1,44 @@
...
-    );
-  }
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { actionCreator } from '../../../../src/redux/game/actions.js';
+
+import Game from './layout.js';
+
+function GameContainer(props) {
```
Comment (guidoprinc) [inline rh-tictactoe/src/app/screens/Game/index.js]: use arrow function

File: rh-tictactoe/src/app/screens/Game/index.js [hunk 8]
```diff
@@ -1,88 +1,44 @@
...
+      status={props.status}
+      stepNumber={props.stepNumber}
+    />
+  );
 }
 
-export default Game;
+GameContainer.propTypes = {
+  handleClick: PropTypes.func.isRequired,
+  history: PropTypes.arrayOf(PropTypes.shape({ squares: PropTypes.oneOf(['X', 'O']) })),
```
Comment (guidoprinc) [inline rh-tictactoe/src/app/screens/Game/index.js]: i want to tell you something when you arrive

File: rh-tictactoe/src/redux/game/actions.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+/* import { calculateWinner } from '../../utils/utils';
+
+ export const nextPlayer = history => dispatch => {
+  if (calculateWinner()) {
+    dispatch({ type: actions.SOMEONE_WON });
+  } else {
+    dispatch({ type: actions.CHANGE_PLAYER });
+  }
+  dispatch(privateActionCreators.makeMove())
+} */
```
Comment (guidoprinc) [inline rh-tictactoe/src/redux/game/actions.js]: delete comments or add a `TODO`

File: rh-tictactoe/src/utils/utils.js (hunk 5 above)
Comment (rodrigoWolox) [reply rh-tictactoe/src/utils/utils.js]: I set it to false but i have some questions about using array.some. But as you said before we can check this later.

File: rh-tictactoe/src/utils/utils.js [hunk 10]
```diff
@@ -1,3 +1,8 @@
+export const actions = {
+  PLAY_TURN: 'PLAY_TURN',
+  JUMP_TO: 'JUMP_TO'
+};
```
Comment (guidoprinc) [inline rh-tictactoe/src/utils/utils.js]: this must go inside corresponding actions file. not utils

File: rh-tictactoe/src/redux/game/reducer.js [hunk 11]
```diff
@@ -9,17 +9,10 @@ const initialState = {
...
-      return utils.makeMove(state, action.squarePosition);
-    case 'JUMP_TO':
-      return utils.jumpTo(state, action.step);
+    case actions.PLAY_TURN:
+      return { ...state, ...action.data };
+    case actions.JUMP_TO:
+      return { ...state, ...action.data };
     default:
       return state;
   }
```
Comment (guidoprinc) [inline rh-tictactoe/src/redux/game/reducer.js]: remember that in the future you can add propTypes here

**Pull request merged**